crates/apps_lib/src/client/tx.rs

```diff
@@ -743,7 +743,18 @@ pub async fn submit_transparent_transfer(
     namada: &impl Namada,
     args: args::TxTransparentTransfer,
 ) -> Result<(), error::Error> {
-    submit_reveal_aux(namada, args.tx.clone(), &args.source).await?;
+    submit_reveal_aux(
+        namada,
+        args.tx.clone(),
+        &args
+            .data
+            .first()
```

Comment:
Do we need to check all sources, not only the first one? Should the source be the same in each transfer?

Reply:
good q - I think this should probably collect a set of unique sources and then iterate over them

Reply:
For the time being I haven't really implemented any kind of support for vectorized transfers in the cli because I believe the ux would be quite bad. So there's no real way to get more than one source in here

Reply:
maybe we could add some check in here then that they are all the same

Reply:
I've added a check on the length of the input data in adbc489ed6bb8934ead1b0b0bfdee095aaf157c4